commit ccc54bd61c768b6a27f9305a0831b76a7b6d706f
Author: Noah Goldstein <goldstein.w.n@gmail.com>
Date:   Wed Jun 29 18:56:18 2022 -0700

    x86: Add missing IS_IN (libc) check to strncmp-sse4_2.S
    
    Was missing to for the multiarch build rtld-strncmp-sse4_2.os was
    being built and exporting symbols:
    
    build/glibc/string/rtld-strncmp-sse4_2.os:
    0000000000000000 T __strncmp_sse42
    
    Introduced in:
    
    commit 11ffcacb64a939c10cfc713746b8ec88837f5c4a
    Author: H.J. Lu <hjl.tools@gmail.com>
    Date:   Wed Jun 21 12:10:50 2017 -0700
    
        x86-64: Implement strcmp family IFUNC selectors in C
    
    (cherry picked from commit 96ac447d915ea5ecef3f9168cc13f4e731349a3b)

diff --git a/sysdeps/x86_64/multiarch/strncmp-sse4_2.S b/sysdeps/x86_64/multiarch/strncmp-sse4_2.S
index 22f51a0dfd2770c9..85dc363bf9d6273d 100644
--- a/sysdeps/x86_64/multiarch/strncmp-sse4_2.S
+++ b/sysdeps/x86_64/multiarch/strncmp-sse4_2.S
@@ -16,6 +16,8 @@
    License along with the GNU C Library; if not, see
    <https://www.gnu.org/licenses/>.  */
 
-#define STRCMP_SSE42 __strncmp_sse42
-#define USE_AS_STRNCMP
-#include "strcmp-sse42.S"
+#if IS_IN (libc)
+# define STRCMP_SSE42 __strncmp_sse42
+# define USE_AS_STRNCMP
+# include "strcmp-sse42.S"
+#endif
